### PR TITLE
Adjusted MSTest tool resolution

### LINKED ISSUE #823: MSTest tool resolution fail if Visual Studio isn't in default location
MSTest tool resolution fail if Visual Studio isn't in default location (program files).

Will submit a PR with suggested solution shortly.

### PR DESCRIPTION
**Pull request by devlead**
* Added fallback to check environment for VisualStudio path if not found in default location
* Fixed parameter typo for ArgumentNullException
* Fixes #823

### CODE REVIEW

**Comment by patriksvensson** [comment]
@devlead I think this needs some tests as well.

**Comment by devlead** [comment]
@patriksvensson fixing that now. As well as the failing test.

File: src/Cake.Common/Tools/MSTest/MSTestRunner.cs [hunk 1]
```diff
@@ -107,5 +116,18 @@ private FilePath GetToolPath(string version)
...
+        {
+            var visualStudioCommonToolsPath = _environment.GetEnvironmentVariable(environmentVariable);
+
+            if (string.IsNullOrWhiteSpace(visualStudioCommonToolsPath))
+            {
+                return null;
+            }
+
+            var root = new DirectoryPath(visualStudioCommonToolsPath).Combine("../IDE");
+            return root.CombineWithFilePath("mstest.exe");
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSTest/MSTestRunner.cs]: Maybe we should use the `PathCollapser` here before returning.

**Comment by devlead** [comment]
@patriksvensson adjusted PR
* Fixed failing test
* Added test for environment variable non standard paths
* Added collapsing the path
* Rebased against latest develop.

**Comment by devlead** [comment]
@patriksvensson rebased against latest develop.

**Comment by patriksvensson** [comment]
@devlead Can you rebase this again?

**Comment by devlead** [comment]
@patriksvensson it should already be rebased.

**Pull request merged**